### PR TITLE
Welcome Page

### PR DESCRIPTION
**Pull request by cassandrale179**
### Changes 

- I open a completely new branch since the old code contained a lot of unrelated things for later features (e.g the Socket). Instead I push all that code to `feature-add-socket`
- The app is now instead running on webpack + rollup using `Svelte` so we can have each component for each view for development. 
- The code changes that need reviews is in the `src/` folder only. All the other files are autogenerated 

## Folder Structure
- live_interviewing/src/extension.ts - main body of the app
- live_interviewing/webviews - Svelte code for the Welcome Page + Sidebar lives here
- out/compiled - JS files are complied here

https://github.com/hatchways/live-interviewing/assets/22923895/1daab9a2-6f04-4561-a750-b546cbfb1b6a

### To do
- [ ] Change the image in the Welcome page to link to a real image for Hatchways product later? (or we can remove the image)
- [ ] Open to wording changes since I'm not 100% sure what steps to put on the Walkthrough page 😅

### CODE REVIEW

File: src/SidebarProvider.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+      localResourceRoots: [this._extensionUri],
+    };
+
+    webviewView.webview.html = this._getHtmlForWebview(webviewView.webview);
+
+    webviewView.webview.onDidReceiveMessage(async (data) => {
+      switch (data.type) {
+        case "inputName": {
+          if (!data.value) {
```
Comment (cassandrale179) [inline src/SidebarProvider.ts:25]: In the future, we can do something with this variable here - which is the candidate's name

File: .wget-hsts [hunk 2]
```diff
@@ -0,0 +1,4 @@
+# HSTS 1.0 Known Hosts database for GNU Wget.
+# Edit at your own risk.
+# <hostname>	<port>	<incl. subdomains>	<created>	<max-age>
+github.com	0	1	1696009051	31536000
```
Comment (skassam21) [inline .wget-hsts]: Should this be ignored?

File: package.json [hunk 3]
```diff
@@ -1,24 +1,117 @@
...
+          },
+          {
+						"id": "openTerminal",
+						"title": "Open Terminal",
+						"description": "Click [here](command:workbench.action.terminal.toggleTerminal) to open terminal. This step will complete once the terminal has been opened.",
+						"completionEvents": [
+							"onView:terminal"
+						],
+            "media": { "image": "media/coding.png", "altText": "Change Settings" }
+					}
```
Comment (skassam21) [inline package.json]: This file needs some prettiering / linting

File: README.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
+# hatchways-live-interviewing README
+
+This is the README for your extension "hatchways-live-interviewing".
```
Comment (ThinkTink) [inline README.md]:
```suggestion
```

File: README.md [hunk 5]
```diff
@@ -0,0 +1,17 @@
+# hatchways-live-interviewing README
```
Comment (ThinkTink) [inline README.md]:
```suggestion
# hatchways-live-interviewing VS Code extension
```

File: package.json [hunk 6]
```diff
@@ -1,24 +1,117 @@
...
+          "icon": "media/hatchways.svg",
+          "contextualTitle": "Hatchways Settings"
+        }
+      ]
+    },
+    "walkthroughs": [
+      {
+        "id": "walkthrough",
+        "title": "Get Started with Hatchways",
+        "description": "Your first steps to setup Hatchways live coding interview",
```
Comment (ThinkTink) [inline package.json]:
```suggestion
        "description": "Your first steps to setup a Hatchways live coding session",
```

File: src/SidebarProvider.ts [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+            return;
+          }
+          vscode.window.showInformationMessage(`Hi ${data.value}, welcome to the interview!`);
```
Comment (ThinkTink) [inline src/SidebarProvider.ts]:
```suggestion
          vscode.window.showInformationMessage(`Hi ${data.value}, welcome to the live coding session!`);
```

**Review by ThinkTink — APPROVED** [review]
Here, can we put the Hatchways logo in the circle?
Screenshot from 2023-10-03 15-34-22

Anything we should be worried about here?
Screenshot from 2023-10-03 15-32-00

**Pull request merged**